infra/minervaStack.js

```diff
@@ -0,0 +1,128 @@
+            assumedBy: new iam.ServicePrincipal( 'lambda.amazonaws.com' ),
+            managedPolicies: [
+                iam.ManagedPolicy.fromAwsManagedPolicyName( 'service-role/AWSLambdaBasicExecutionRole' ),
+            ],
+        } )
+
+        // Create Lambda functions
+        const slackCommandsSync = new NodejsFunction( this, 'SlackCommandsSync', {
+            ...lambdaFnProps,
+            handler: 'index.slack_commands_sync',
+            functionName: `minerva-${ deployEnv }-slackCommandsSync`,
+            role: lambdaWithExecuteRole,
+        } );
+
+        const slackCommandsAsync = new NodejsFunction( this, 'SlackCommandsAsync', {
+            ...lambdaFnProps,
+            handler: 'index.slack_commands_async',
+            functionName: `minerva-${ deployEnv }-slackCommandsAsync`,
+            role: lambdaRole,
+        } );
+
+        const interactivitySync = new NodejsFunction( this, 'InteractivitySync', {
+            ...lambdaFnProps,
+            handler: 'index.interactivity_sync',
+            functionName: `minerva-${ deployEnv }-interactivitySync`,
+            role: lambdaWithExecuteRole,
+        } );
+
+        const interactivityAsync = new NodejsFunction( this, 'InteractivityAsync', {
+            ...lambdaFnProps,
+            handler: 'index.interactivity_async',
+            functionName: `minerva-${ deployEnv }-interactivityAsync`,
+            role: lambdaRole,
+        } );
+
+        const scheduled = new NodejsFunction( this, 'Scheduled', {
+            ...lambdaFnProps,
+            handler: 'index.scheduled',
+            functionName: `minerva-${ deployEnv }-scheduled`,
+            role: lambdaWithExecuteRole,
+        });
+
+        // Create API Gateway
+        const minervaApi = new apiGateway.RestApi( this, 'MinervaApi', {
+            deployOptions: {stageName: deployEnv},
+        } )
+
+        // All Slack lambda functions are under the same "slack" route
+        const slackRoute = minervaApi.root.addResource('slack');
+        // Attach the Lambda functions to the Slack route
+        slackRoute.addResource('commands-sync').addMethod('POST', new apiGateway.LambdaIntegration(slackCommandsSync));
+        slackRoute.addResource('commands-async').addMethod('POST', new apiGateway.LambdaIntegration(slackCommandsAsync));
+        slackRoute.addResource('interactivity-sync').addMethod('POST', new apiGateway.LambdaIntegration(interactivitySync));
+        slackRoute.addResource('interactivity-async').addMethod('POST', new apiGateway.LambdaIntegration(interactivityAsync));
+        slackRoute.addResource('scheduled').addMethod('POST', new apiGateway.LambdaIntegration(scheduled));
+
+        // Create an EventBridge rule to trigger the lambda functions every minute, keeping the lambda functions warm
+        const keepLambdaWarmRule = new awsEvents.Rule( this, 'KeepLambdaWarmRule', {
+            schedule: awsEvents.Schedule.rate( cdk.Duration.minutes( 1 ) ),
+            targets: [
```

Comment:
I would like to DRY this and the above `slackRoute` configurations, but maybe that's for a later PR